numba/cuda/tests/cudapy/test_sync.py

```diff
@@ -71,6 +90,30 @@ def test_dyn_shared_memory(self):
         compiled[1, shape, 0, ary.size * 4](ary)
         self.assertTrue(np.all(ary == 2 * np.arange(ary.size, dtype=np.int32)))
 
+    def test_threadfence_codegen(self):
+        compiled = cuda.jit("void(int32[:])")(use_threadfence)
+        ary = np.zeros(10, dtype=np.int32)
+        compiled[1, 1](ary)
```

Comment:
My intention was really just to test code generation side of things: making sure that the right PTX instruction was emitted.  That's why I postfix the test name with "_codegen", but I should have documented the intention.  I cannot think of a good way to test the feature.  And, I don't think it is necessary to test the memfence runtime behavior as it should be guaranteed by the HW as long as it is emitted.